server/libs/infra/src/entities/partner.entity.ts

```diff
@@ -0,0 +1,21 @@
+import { CreateDateColumn, Entity, ManyToOne, PrimaryGeneratedColumn, UpdateDateColumn } from 'typeorm';
+
+import { UserEntity } from './user.entity';
+
+@Entity('partners')
+export class PartnerEntity {
+  @PrimaryGeneratedColumn('uuid')
+  id!: string;
+
+  @ManyToOne(() => UserEntity, { onDelete: 'CASCADE' })
+  sharedBy!: string;
+
+  @ManyToOne(() => UserEntity, { onDelete: 'CASCADE' })
+  sharedWith!: string;
```

Comment:
This is basically an many to many intersection table between user and user. I don't think we need the id column here, and I don't think it is used anywhere, is it? The shared with/by combination could be a joint primary column, which is how intersection tables often are configured (both columns have `@PrimaryColumn()` decorators). I believe right now `UserA` could re-share with `UserB` twice and the database would have no problem with it. Since we don't need that specific functionality I think I'd prefer using two columns for the primary key instead of a uuid.

I've never used`@ManyToX` relationships with the type `string` oppose to `Entity`. Does this just return the uuid of the user oppose to the user entity, or does it return the user and the type is just wrong?